homeassistant/components/neato/vacuum.py

```diff
@@ -185,13 +185,16 @@ def update(self):
             self._mapdata[self._robot_serial]['maps'][0]['run_charge_at_end'])
 
         if self._robot_has_map:
+            robot_map_id = {}
             if self._state['availableServices']['maps'] != "basic-1":
                 if self._robot_maps[self._robot_serial]:
-                    robot_map_id = (
-                        self._robot_maps[self._robot_serial][0]['id'])
+                    allmaps = self._robot_maps[self._robot_serial]
+                    for maps in allmaps:
+                        robot_map_id[allmaps.index(maps)] = maps['id']
 
-                    self._robot_boundaries = self.robot.get_map_boundaries(
-                        robot_map_id).json()
+                    for map_id, neatomap in robot_map_id.items():
```

Comment:
Your editor was correct. :smile:
This line should be:
```py
for neatomap in robot_map_id.values():
    ...
```

Reply:
Maybe we don't need a dict, but a list would be enough?

Reply:
Why can't we iterate `allmaps` directly?

Reply:
I am not sure, the purpose of this fix is to make sure we grab all zones from all floorplans.  The old code was grabbing the code from the first floorplan (hence the `[0]` below) so only those zones were being found.

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/neato/vacuum.py#L191

FWIW all of the code changes you recommended still allow the service call to work as expected :)